### PR TITLE
Fixed UI freeze on typing a string literal inside some complex function

### PR DESCRIPTION
**Pull request by vlad20012**
Related to #2733

### CODE REVIEW

File: src/main/kotlin/org/rust/ide/injected/RsRegExpInjector.kt [hunk 1]
```diff
@@ -39,10 +39,16 @@ class RsRegExpInjector : MultiHostInjector {
...
-        val fn = ((context.parent?.parent as? RsCallExpr)?.expr as? RsPathExpr)
-            ?.path?.reference?.resolve() as? RsFunction
-            ?: return false
-        return fn.name == "new" && fn.implTypeName == "Regex"
+//        val fn = ((context.parent?.parent as? RsCallExpr)?.expr as? RsPathExpr)
+//            ?.path?.reference?.resolve() as? RsFunction
+//            ?: return false
+//        return fn.name == "new" && fn.implTypeName == "Regex"
+
+        // We switched to this dumb implementation because this code sometimes be
```
Comment (Undin) [inline src/main/kotlin/org/rust/ide/injected/RsRegExpInjector.kt]: "this code sometimes be" -> "this code sometimes can be"

File: src/main/kotlin/org/rust/ide/injected/RsRegExpInjector.kt [hunk 2]
```diff
@@ -39,10 +39,16 @@ class RsRegExpInjector : MultiHostInjector {
...
-            ?.path?.reference?.resolve() as? RsFunction
-            ?: return false
-        return fn.name == "new" && fn.implTypeName == "Regex"
+//        val fn = ((context.parent?.parent as? RsCallExpr)?.expr as? RsPathExpr)
+//            ?.path?.reference?.resolve() as? RsFunction
+//            ?: return false
+//        return fn.name == "new" && fn.implTypeName == "Regex"
+
+        // We switched to this dumb implementation because this code is sometimes
+        // called from EDT and invoking `reference.resolve()` can freeze the UI.
```
Comment (ortem) [inline src/main/kotlin/org/rust/ide/injected/RsRegExpInjector.kt]: Maybe `<...>called from EDT, and invoking<...>`?

**Pull request merged**